stm32-modules/include/common/core/at24c0xc.hpp

```diff
@@ -0,0 +1,146 @@
+/**
+ * @file at24c0xc.hpp
+ * @brief Implements a generic driver for the AT24C0xC EEPROM IC's.
+ */
+
+#pragma once
+
+#include <array>
+#include <cstring>  // For memcpy
+
+#include "core/bit_utils.hpp"
+
+// Namespace for AT24C01C / AT24C02C driver
+namespace at24c0xc {
+
+static constexpr const size_t PAGE_LENGTH = 8;
+
+template <typename Policy>
+concept AT24C0xC_Policy = requires(Policy &policy, uint8_t addr,
```

Comment:
looks good; I tend to personally prefer iterator-sentinel pairs rather than iterator-size, and I'd love to find a way to make it not require specifically array iterator,s but this definitely is good

Reply:
I was going to initially go with that but the issue is passing the buffer to the HAL I2C driver, which takes a pointer + length. With the iterator-sentinel pair the policy would end up just having to call `std::distance` on the two iterators to get the length.

I've got some i2c refactor stuff ticketed in Jira, will make sure to take another pass at this with that work